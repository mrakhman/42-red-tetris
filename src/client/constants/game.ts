export const maxPlayersLimit = 4;
