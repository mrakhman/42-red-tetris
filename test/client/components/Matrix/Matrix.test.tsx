import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import Matrix from '/client/components/Matrix';
import { blankMatrix, blockMatrix } from '/client/constants/tetriminos';
import { blockCreate } from '/client/controllers/blockControllers';


describe('Snapshots', () => {
  it('should render blank Matrix with all props', () => {
    const props = {
      block: blockCreate({ type: 'T', pos: [0, 0] }),
      matrix: blankMatrix,
      style: { marginLeft: 10, borderWidth: 0, marginVertical: 10 },
      isSpectrum: undefined,
    };

    const wrapper = shallow(<Matrix {...props} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should render next block Matrix with all props', () => {
    const props = {
      block: blockCreate({ type: 'T', pos: [0, 0] }),
      matrix: blockMatrix,
      style: { marginLeft: 10, borderWidth: 0, marginVertical: 10 },
      isSpectrum: undefined,
    };

    const wrapper = shallow(<Matrix {...props} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should render spectrum Matrix', () => {
    const props = {
      block: blockCreate({ type: 'O', pos: [-10, -10] }),
      matrix: blankMatrix,
      style: undefined,
      isSpectrum: true,
    };

    const wrapper = shallow(<Matrix {...props} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
