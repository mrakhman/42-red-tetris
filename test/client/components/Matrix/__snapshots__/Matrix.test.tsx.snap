// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshots should render blank Matrix with all props 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#9ead86",
        "borderColor": "black",
        "borderWidth": 1,
        "justifyContent": "space-between",
        "padding": 2,
      },
      Object {
        "height": 440,
      },
      Object {
        "borderWidth": 0,
        "marginLeft": 10,
        "marginVertical": 10,
      },
    ]
  }
>
  <View
    key="0"
    style={
      Object {
        "flexDirection": "row",
        "justifyContent": "space-between",
        "width": 220,
      }
    }
  >
    <Block
      key="0"
      value={0}
    />
    <Block
      key="1"
      value={1}
    />
    <Block
      key="2"
      value={0}
    />
    <Block
      key="3"
      value={0}
    />
    <Block
      key="4"
      value={0}
    />
    <Block
      key="5"
      value={0}
    />
    <Block
      key="6"
      value={0}
    />
    <Block
      key="7"
      value={0}
    />
    <Block
      key="8"
      value={0}
    />
    <Block
      key="9"
      value={0}
    />
  </View>
  <View
    key="1"
    style={
      Object {
        "flexDirection": "row",
        "justifyContent": "space-between",
        "width": 220,
      }
    }
  >
    <Block
      key="0"
      value={1}
    />
    <Block
      key="1"
      value={1}
    />
    <Block
      key="2"
      value={1}
    />
    <Block
      key="3"
      value={0}
    />
    <Block
      key="4"
      value={0}
    />
    <Block
      key="5"
      value={0}
    />
    <Block
      key="6"
      value={0}
    />
    <Block
      key="7"
      value={0}
    />
    <Block
      key="8"
      value={0}
    />
    <Block
      key="9"
      value={0}
    />
  </View>
  <View
    key="2"
    style={
      Object {
        "flexDirection": "row",
        "justifyContent": "space-between",
        "width": 220,
      }
    }
  >
    <Block
      key="0"
      value={0}
    />
    <Block
      key="1"
      value={0}
    />
    <Block
      key="2"
      value={0}
    />
    <Block
      key="3"
      value={0}
    />
    <Block
      key="4"
      value={0}
    />
    <Block
      key="5"
      value={0}
    />
    <Block
      key="6"
      value={0}
    />
    <Block
      key="7"
      value={0}
    />
    <Block
      key="8"
      value={0}
    />
    <Block
      key="9"
      value={0}
    />
  </View>
  <View
    key="3"
    style={
      Object {
        "flexDirection": "row",
        "justifyContent": "space-between",
        "width": 220,
      }
    }
  >
    <Block
      key="0"
      value={0}
    />
    <Block
      key="1"
      value={0}
    />
    <Block
      key="2"
      value={0}
    />
    <Block
      key="3"
      value={0}
    />
    <Block
      key="4"
      value={0}
    />
    <Block
      key="5"
      value={0}
    />
    <Block
      key="6"
      value={0}
    />
    <Block
      key="7"
      value={0}
    />
    <Block
      key="8"
      value={0}
    />
    <Block
      key="9"
      value={0}
    />
  </View>
  <View
    key="4"
    style={
      Object {
        "flexDirection": "row",
        "justifyContent": "space-between",
        "width": 220,
      }
    }
  >
    <Block
      key="0"
      value={0}
    />
    <Block
      key="1"
      value={0}
    />
    <Block
      key="2"
      value={0}
    />
    <Block
      key="3"
      value={0}
    />
    <Block
      key="4"
      value={0}
    />
    <Block
      key="5"
      value={0}
    />
    <Block
      key="6"
      value={0}
    />
    <Block
      key="7"
      value={0}
    />
    <Block
      key="8"
      value={0}
    />
    <Block
      key="9"
      value={0}
    />
  </View>
  <View
    key="5"
    style={
      Object {
        "flexDirection": "row",
        "justifyContent": "space-between",
        "width": 220,
      }
    }
  >
    <Block
      key="0"
      value={0}
    />
    <Block
      key="1"
      value={0}
    />
    <Block
      key="2"
      value={0}
    />
    <Block
      key="3"
      value={0}
    />
    <Block
      key="4"
      value={0}
    />
    <Block
      key="5"
      value={0}
    />
    <Block
      key="6"
      value={0}
    />
    <Block
      key="7"
      value={0}
    />
    <Block
      key="8"
      value={0}
    />
    <Block
      key="9"
      value={0}
    />
  </View>
  <View
    key="6"
    style={
      Object {
        "flexDirection": "row",
        "justifyContent": "space-between",
        "width": 220,
      }
    }
  >
    <Block
      key="0"
      value={0}
    />
    <Block
      key="1"
      value={0}
    />
    <Block
      key="2"
      value={0}
    />
    <Block
      key="3"
      value={0}
    />
    <Block
      key="4"
      value={0}
    />
    <Block
      key="5"
      value={0}
    />
    <Block
      key="6"
      value={0}
    />
    <Block
      key="7"
      value={0}
    />
    <Block
      key="8"
      value={0}
    />
    <Block
      key="9"
      value={0}
    />
  </View>
  <View
    key="7"
    style={
      Object {
        "flexDirection": "row",
        "justifyContent": "space-between",
        "width": 220,
      }
    }
  >
    <Block
      key="0"
      value={0}
    />
    <Block
      key="1"
      value={0}
    />
    <Block
      key="2"
      value={0}
    />
    <Block
      key="3"
      value={0}
    />
    <Block
      key="4"
      value={0}
    />
    <Block
      key="5"
      value={0}
    />
    <Block
      key="6"
      value={0}
    />
    <Block
      key="7"
      value={0}
    />
    <Block
      key="8"
      value={0}
    />
    <Block
      key="9"
      value={0}
    />
  </View>
  <View
    key="8"
    style={
      Object {
        "flexDirection": "row",
        "justifyContent": "space-between",
        "width": 220,
      }
    }
  >
    <Block
      key="0"
      value={0}
    />
    <Block
      key="1"
      value={0}
    />
    <Block
      key="2"
      value={0}
    />
    <Block
      key="3"
      value={0}
    />
    <Block
      key="4"
      value={0}
    />
    <Block
      key="5"
      value={0}
    />
    <Block
      key="6"
      value={0}
    />
    <Block
      key="7"
      value={0}
    />
    <Block
      key="8"
      value={0}
    />
    <Block
      key="9"
      value={0}
    />
  </View>
  <View
    key="9"
    style={
      Object {
        "flexDirection": "row",
        "justifyContent": "space-between",
        "width": 220,
      }
    }
  >
    <Block
      key="0"
      value={0}
    />
    <Block
      key="1"
      value={0}
    />
    <Block
      key="2"
      value={0}
    />
    <Block
      key="3"
      value={0}
    />
    <Block
      key="4"
      value={0}
    />
    <Block
      key="5"
      value={0}
    />
    <Block
      key="6"
      value={0}
    />
    <Block
      key="7"
      value={0}
    />
    <Block
      key="8"
      value={0}
    />
    <Block
      key="9"
      value={0}
    />
  </View>
  <View
    key="10"
    style={
      Object {
        "flexDirection": "row",
        "justifyContent": "space-between",
        "width": 220,
      }
    }
  >
    <Block
      key="0"
      value={0}
    />
    <Block
      key="1"
      value={0}
    />
    <Block
      key="2"
      value={0}
    />
    <Block
      key="3"
      value={0}
    />
    <Block
      key="4"
      value={0}
    />
    <Block
      key="5"
      value={0}
    />
    <Block
      key="6"
      value={0}
    />
    <Block
      key="7"
      value={0}
    />
    <Block
      key="8"
      value={0}
    />
    <Block
      key="9"
      value={0}
    />
  </View>
  <View
    key="11"
    style={
      Object {
        "flexDirection": "row",
        "justifyContent": "space-between",
        "width": 220,
      }
    }
  >
    <Block
      key="0"
      value={0}
    />
    <Block
      key="1"
      value={0}
    />
    <Block
      key="2"
      value={0}
    />
    <Block
      key="3"
      value={0}
    />
    <Block
      key="4"
      value={0}
    />
    <Block
      key="5"
      value={0}
    />
    <Block
      key="6"
      value={0}
    />
    <Block
      key="7"
      value={0}
    />
    <Block
      key="8"
      value={0}
    />
    <Block
      key="9"
      value={0}
    />
  </View>
  <View
    key="12"
    style={
      Object {
        "flexDirection": "row",
        "justifyContent": "space-between",
        "width": 220,
      }
    }
  >
    <Block
      key="0"
      value={0}
    />
    <Block
      key="1"
      value={0}
    />
    <Block
      key="2"
      value={0}
    />
    <Block
      key="3"
      value={0}
    />
    <Block
      key="4"
      value={0}
    />
    <Block
      key="5"
      value={0}
    />
    <Block
      key="6"
      value={0}
    />
    <Block
      key="7"
      value={0}
    />
    <Block
      key="8"
      value={0}
    />
    <Block
      key="9"
      value={0}
    />
  </View>
  <View
    key="13"
    style={
      Object {
        "flexDirection": "row",
        "justifyContent": "space-between",
        "width": 220,
      }
    }
  >
    <Block
      key="0"
      value={0}
    />
    <Block
      key="1"
      value={0}
    />
    <Block
      key="2"
      value={0}
    />
    <Block
      key="3"
      value={0}
    />
    <Block
      key="4"
      value={0}
    />
    <Block
      key="5"
      value={0}
    />
    <Block
      key="6"
      value={0}
    />
    <Block
      key="7"
      value={0}
    />
    <Block
      key="8"
      value={0}
    />
    <Block
      key="9"
      value={0}
    />
  </View>
  <View
    key="14"
    style={
      Object {
        "flexDirection": "row",
        "justifyContent": "space-between",
        "width": 220,
      }
    }
  >
    <Block
      key="0"
      value={0}
    />
    <Block
      key="1"
      value={0}
    />
    <Block
      key="2"
      value={0}
    />
    <Block
      key="3"
      value={0}
    />
    <Block
      key="4"
      value={0}
    />
    <Block
      key="5"
      value={0}
    />
    <Block
      key="6"
      value={0}
    />
    <Block
      key="7"
      value={0}
    />
    <Block
      key="8"
      value={0}
    />
    <Block
      key="9"
      value={0}
    />
  </View>
  <View
    key="15"
    style={
      Object {
        "flexDirection": "row",
        "justifyContent": "space-between",
        "width": 220,
      }
    }
  >
    <Block
      key="0"
      value={0}
    />
    <Block
      key="1"
      value={0}
    />
    <Block
      key="2"
      value={0}
    />
    <Block
      key="3"
      value={0}
    />
    <Block
      key="4"
      value={0}
    />
    <Block
      key="5"
      value={0}
    />
    <Block
      key="6"
      value={0}
    />
    <Block
      key="7"
      value={0}
    />
    <Block
      key="8"
      value={0}
    />
    <Block
      key="9"
      value={0}
    />
  </View>
  <View
    key="16"
    style={
      Object {
        "flexDirection": "row",
        "justifyContent": "space-between",
        "width": 220,
      }
    }
  >
    <Block
      key="0"
      value={0}
    />
    <Block
      key="1"
      value={0}
    />
    <Block
      key="2"
      value={0}
    />
    <Block
      key="3"
      value={0}
    />
    <Block
      key="4"
      value={0}
    />
    <Block
      key="5"
      value={0}
    />
    <Block
      key="6"
      value={0}
    />
    <Block
      key="7"
      value={0}
    />
    <Block
      key="8"
      value={0}
    />
    <Block
      key="9"
      value={0}
    />
  </View>
  <View
    key="17"
    style={
      Object {
        "flexDirection": "row",
        "justifyContent": "space-between",
        "width": 220,
      }
    }
  >
    <Block
      key="0"
      value={0}
    />
    <Block
      key="1"
      value={0}
    />
    <Block
      key="2"
      value={0}
    />
    <Block
      key="3"
      value={0}
    />
    <Block
      key="4"
      value={0}
    />
    <Block
      key="5"
      value={0}
    />
    <Block
      key="6"
      value={0}
    />
    <Block
      key="7"
      value={0}
    />
    <Block
      key="8"
      value={0}
    />
    <Block
      key="9"
      value={0}
    />
  </View>
  <View
    key="18"
    style={
      Object {
        "flexDirection": "row",
        "justifyContent": "space-between",
        "width": 220,
      }
    }
  >
    <Block
      key="0"
      value={0}
    />
    <Block
      key="1"
      value={0}
    />
    <Block
      key="2"
      value={0}
    />
    <Block
      key="3"
      value={0}
    />
    <Block
      key="4"
      value={0}
    />
    <Block
      key="5"
      value={0}
    />
    <Block
      key="6"
      value={0}
    />
    <Block
      key="7"
      value={0}
    />
    <Block
      key="8"
      value={0}
    />
    <Block
      key="9"
      value={0}
    />
  </View>
  <View
    key="19"
    style={
      Object {
        "flexDirection": "row",
        "justifyContent": "space-between",
        "width": 220,
      }
    }
  >
    <Block
      key="0"
      value={0}
    />
    <Block
      key="1"
      value={0}
    />
    <Block
      key="2"
      value={0}
    />
    <Block
      key="3"
      value={0}
    />
    <Block
      key="4"
      value={0}
    />
    <Block
      key="5"
      value={0}
    />
    <Block
      key="6"
      value={0}
    />
    <Block
      key="7"
      value={0}
    />
    <Block
      key="8"
      value={0}
    />
    <Block
      key="9"
      value={0}
    />
  </View>
</View>
`;

exports[`Snapshots should render next block Matrix with all props 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#9ead86",
        "borderColor": "black",
        "borderWidth": 1,
        "justifyContent": "space-between",
        "padding": 2,
      },
      Object {
        "height": 44,
      },
      Object {
        "borderWidth": 0,
        "marginLeft": 10,
        "marginVertical": 10,
      },
    ]
  }
>
  <View
    key="0"
    style={
      Object {
        "flexDirection": "row",
        "justifyContent": "space-between",
        "width": 88,
      }
    }
  >
    <Block
      key="0"
      value={0}
    />
    <Block
      key="1"
      value={1}
    />
    <Block
      key="2"
      value={0}
    />
    <Block
      key="3"
      value={0}
    />
  </View>
  <View
    key="1"
    style={
      Object {
        "flexDirection": "row",
        "justifyContent": "space-between",
        "width": 88,
      }
    }
  >
    <Block
      key="0"
      value={1}
    />
    <Block
      key="1"
      value={1}
    />
    <Block
      key="2"
      value={1}
    />
    <Block
      key="3"
      value={0}
    />
  </View>
</View>
`;

exports[`Snapshots should render spectrum Matrix 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#9ead86",
        "borderColor": "black",
        "borderWidth": 1,
        "justifyContent": "space-between",
        "padding": 2,
      },
      Object {
        "height": 160,
      },
      undefined,
    ]
  }
>
  <View
    key="0"
    style={
      Object {
        "flexDirection": "row",
        "justifyContent": "space-between",
        "width": 80,
      }
    }
  >
    <Block
      isSpectrum={true}
      key="0"
      value={0}
    />
    <Block
      isSpectrum={true}
      key="1"
      value={0}
    />
    <Block
      isSpectrum={true}
      key="2"
      value={0}
    />
    <Block
      isSpectrum={true}
      key="3"
      value={0}
    />
    <Block
      isSpectrum={true}
      key="4"
      value={0}
    />
    <Block
      isSpectrum={true}
      key="5"
      value={0}
    />
    <Block
      isSpectrum={true}
      key="6"
      value={0}
    />
    <Block
      isSpectrum={true}
      key="7"
      value={0}
    />
    <Block
      isSpectrum={true}
      key="8"
      value={0}
    />
    <Block
      isSpectrum={true}
      key="9"
      value={0}
    />
  </View>
  <View
    key="1"
    style={
      Object {
        "flexDirection": "row",
        "justifyContent": "space-between",
        "width": 80,
      }
    }
  >
    <Block
      isSpectrum={true}
      key="0"
      value={0}
    />
    <Block
      isSpectrum={true}
      key="1"
      value={0}
    />
    <Block
      isSpectrum={true}
      key="2"
      value={0}
    />
    <Block
      isSpectrum={true}
      key="3"
      value={0}
    />
    <Block
      isSpectrum={true}
      key="4"
      value={0}
    />
    <Block
      isSpectrum={true}
      key="5"
      value={0}
    />
    <Block
      isSpectrum={true}
      key="6"
      value={0}
    />
    <Block
      isSpectrum={true}
      key="7"
      value={0}
    />
    <Block
      isSpectrum={true}
      key="8"
      value={0}
    />
    <Block
      isSpectrum={true}
      key="9"
      value={0}
    />
  </View>
  <View
    key="2"
    style={
      Object {
        "flexDirection": "row",
        "justifyContent": "space-between",
        "width": 80,
      }
    }
  >
    <Block
      isSpectrum={true}
      key="0"
      value={0}
    />
    <Block
      isSpectrum={true}
      key="1"
      value={0}
    />
    <Block
      isSpectrum={true}
      key="2"
      value={0}
    />
    <Block
      isSpectrum={true}
      key="3"
      value={0}
    />
    <Block
      isSpectrum={true}
      key="4"
      value={0}
    />
    <Block
      isSpectrum={true}
      key="5"
      value={0}
    />
    <Block
      isSpectrum={true}
      key="6"
      value={0}
    />
    <Block
      isSpectrum={true}
      key="7"
      value={0}
    />
    <Block
      isSpectrum={true}
      key="8"
      value={0}
    />
    <Block
      isSpectrum={true}
      key="9"
      value={0}
    />
  </View>
  <View
    key="3"
    style={
      Object {
        "flexDirection": "row",
        "justifyContent": "space-between",
        "width": 80,
      }
    }
  >
    <Block
      isSpectrum={true}
      key="0"
      value={0}
    />
    <Block
      isSpectrum={true}
      key="1"
      value={0}
    />
    <Block
      isSpectrum={true}
      key="2"
      value={0}
    />
    <Block
      isSpectrum={true}
      key="3"
      value={0}
    />
    <Block
      isSpectrum={true}
      key="4"
      value={0}
    />
    <Block
      isSpectrum={true}
      key="5"
      value={0}
    />
    <Block
      isSpectrum={true}
      key="6"
      value={0}
    />
    <Block
      isSpectrum={true}
      key="7"
      value={0}
    />
    <Block
      isSpectrum={true}
      key="8"
      value={0}
    />
    <Block
      isSpectrum={true}
      key="9"
      value={0}
    />
  </View>
  <View
    key="4"
    style={
      Object {
        "flexDirection": "row",
        "justifyContent": "space-between",
        "width": 80,
      }
    }
  >
    <Block
      isSpectrum={true}
      key="0"
      value={0}
    />
    <Block
      isSpectrum={true}
      key="1"
      value={0}
    />
    <Block
      isSpectrum={true}
      key="2"
      value={0}
    />
    <Block
      isSpectrum={true}
      key="3"
      value={0}
    />
    <Block
      isSpectrum={true}
      key="4"
      value={0}
    />
    <Block
      isSpectrum={true}
      key="5"
      value={0}
    />
    <Block
      isSpectrum={true}
      key="6"
      value={0}
    />
    <Block
      isSpectrum={true}
      key="7"
      value={0}
    />
    <Block
      isSpectrum={true}
      key="8"
      value={0}
    />
    <Block
      isSpectrum={true}
      key="9"
      value={0}
    />
  </View>
  <View
    key="5"
    style={
      Object {
        "flexDirection": "row",
        "justifyContent": "space-between",
        "width": 80,
      }
    }
  >
    <Block
      isSpectrum={true}
      key="0"
      value={0}
    />
    <Block
      isSpectrum={true}
      key="1"
      value={0}
    />
    <Block
      isSpectrum={true}
      key="2"
      value={0}
    />
    <Block
      isSpectrum={true}
      key="3"
      value={0}
    />
    <Block
      isSpectrum={true}
      key="4"
      value={0}
    />
    <Block
      isSpectrum={true}
      key="5"
      value={0}
    />
    <Block
      isSpectrum={true}
      key="6"
      value={0}
    />
    <Block
      isSpectrum={true}
      key="7"
      value={0}
    />
    <Block
      isSpectrum={true}
      key="8"
      value={0}
    />
    <Block
      isSpectrum={true}
      key="9"
      value={0}
    />
  </View>
  <View
    key="6"
    style={
      Object {
        "flexDirection": "row",
        "justifyContent": "space-between",
        "width": 80,
      }
    }
  >
    <Block
      isSpectrum={true}
      key="0"
      value={0}
    />
    <Block
      isSpectrum={true}
      key="1"
      value={0}
    />
    <Block
      isSpectrum={true}
      key="2"
      value={0}
    />
    <Block
      isSpectrum={true}
      key="3"
      value={0}
    />
    <Block
      isSpectrum={true}
      key="4"
      value={0}
    />
    <Block
      isSpectrum={true}
      key="5"
      value={0}
    />
    <Block
      isSpectrum={true}
      key="6"
      value={0}
    />
    <Block
      isSpectrum={true}
      key="7"
      value={0}
    />
    <Block
      isSpectrum={true}
      key="8"
      value={0}
    />
    <Block
      isSpectrum={true}
      key="9"
      value={0}
    />
  </View>
  <View
    key="7"
    style={
      Object {
        "flexDirection": "row",
        "justifyContent": "space-between",
        "width": 80,
      }
    }
  >
    <Block
      isSpectrum={true}
      key="0"
      value={0}
    />
    <Block
      isSpectrum={true}
      key="1"
      value={0}
    />
    <Block
      isSpectrum={true}
      key="2"
      value={0}
    />
    <Block
      isSpectrum={true}
      key="3"
      value={0}
    />
    <Block
      isSpectrum={true}
      key="4"
      value={0}
    />
    <Block
      isSpectrum={true}
      key="5"
      value={0}
    />
    <Block
      isSpectrum={true}
      key="6"
      value={0}
    />
    <Block
      isSpectrum={true}
      key="7"
      value={0}
    />
    <Block
      isSpectrum={true}
      key="8"
      value={0}
    />
    <Block
      isSpectrum={true}
      key="9"
      value={0}
    />
  </View>
  <View
    key="8"
    style={
      Object {
        "flexDirection": "row",
        "justifyContent": "space-between",
        "width": 80,
      }
    }
  >
    <Block
      isSpectrum={true}
      key="0"
      value={0}
    />
    <Block
      isSpectrum={true}
      key="1"
      value={0}
    />
    <Block
      isSpectrum={true}
      key="2"
      value={0}
    />
    <Block
      isSpectrum={true}
      key="3"
      value={0}
    />
    <Block
      isSpectrum={true}
      key="4"
      value={0}
    />
    <Block
      isSpectrum={true}
      key="5"
      value={0}
    />
    <Block
      isSpectrum={true}
      key="6"
      value={0}
    />
    <Block
      isSpectrum={true}
      key="7"
      value={0}
    />
    <Block
      isSpectrum={true}
      key="8"
      value={0}
    />
    <Block
      isSpectrum={true}
      key="9"
      value={0}
    />
  </View>
  <View
    key="9"
    style={
      Object {
        "flexDirection": "row",
        "justifyContent": "space-between",
        "width": 80,
      }
    }
  >
    <Block
      isSpectrum={true}
      key="0"
      value={0}
    />
    <Block
      isSpectrum={true}
      key="1"
      value={0}
    />
    <Block
      isSpectrum={true}
      key="2"
      value={0}
    />
    <Block
      isSpectrum={true}
      key="3"
      value={0}
    />
    <Block
      isSpectrum={true}
      key="4"
      value={0}
    />
    <Block
      isSpectrum={true}
      key="5"
      value={0}
    />
    <Block
      isSpectrum={true}
      key="6"
      value={0}
    />
    <Block
      isSpectrum={true}
      key="7"
      value={0}
    />
    <Block
      isSpectrum={true}
      key="8"
      value={0}
    />
    <Block
      isSpectrum={true}
      key="9"
      value={0}
    />
  </View>
  <View
    key="10"
    style={
      Object {
        "flexDirection": "row",
        "justifyContent": "space-between",
        "width": 80,
      }
    }
  >
    <Block
      isSpectrum={true}
      key="0"
      value={0}
    />
    <Block
      isSpectrum={true}
      key="1"
      value={0}
    />
    <Block
      isSpectrum={true}
      key="2"
      value={0}
    />
    <Block
      isSpectrum={true}
      key="3"
      value={0}
    />
    <Block
      isSpectrum={true}
      key="4"
      value={0}
    />
    <Block
      isSpectrum={true}
      key="5"
      value={0}
    />
    <Block
      isSpectrum={true}
      key="6"
      value={0}
    />
    <Block
      isSpectrum={true}
      key="7"
      value={0}
    />
    <Block
      isSpectrum={true}
      key="8"
      value={0}
    />
    <Block
      isSpectrum={true}
      key="9"
      value={0}
    />
  </View>
  <View
    key="11"
    style={
      Object {
        "flexDirection": "row",
        "justifyContent": "space-between",
        "width": 80,
      }
    }
  >
    <Block
      isSpectrum={true}
      key="0"
      value={0}
    />
    <Block
      isSpectrum={true}
      key="1"
      value={0}
    />
    <Block
      isSpectrum={true}
      key="2"
      value={0}
    />
    <Block
      isSpectrum={true}
      key="3"
      value={0}
    />
    <Block
      isSpectrum={true}
      key="4"
      value={0}
    />
    <Block
      isSpectrum={true}
      key="5"
      value={0}
    />
    <Block
      isSpectrum={true}
      key="6"
      value={0}
    />
    <Block
      isSpectrum={true}
      key="7"
      value={0}
    />
    <Block
      isSpectrum={true}
      key="8"
      value={0}
    />
    <Block
      isSpectrum={true}
      key="9"
      value={0}
    />
  </View>
  <View
    key="12"
    style={
      Object {
        "flexDirection": "row",
        "justifyContent": "space-between",
        "width": 80,
      }
    }
  >
    <Block
      isSpectrum={true}
      key="0"
      value={0}
    />
    <Block
      isSpectrum={true}
      key="1"
      value={0}
    />
    <Block
      isSpectrum={true}
      key="2"
      value={0}
    />
    <Block
      isSpectrum={true}
      key="3"
      value={0}
    />
    <Block
      isSpectrum={true}
      key="4"
      value={0}
    />
    <Block
      isSpectrum={true}
      key="5"
      value={0}
    />
    <Block
      isSpectrum={true}
      key="6"
      value={0}
    />
    <Block
      isSpectrum={true}
      key="7"
      value={0}
    />
    <Block
      isSpectrum={true}
      key="8"
      value={0}
    />
    <Block
      isSpectrum={true}
      key="9"
      value={0}
    />
  </View>
  <View
    key="13"
    style={
      Object {
        "flexDirection": "row",
        "justifyContent": "space-between",
        "width": 80,
      }
    }
  >
    <Block
      isSpectrum={true}
      key="0"
      value={0}
    />
    <Block
      isSpectrum={true}
      key="1"
      value={0}
    />
    <Block
      isSpectrum={true}
      key="2"
      value={0}
    />
    <Block
      isSpectrum={true}
      key="3"
      value={0}
    />
    <Block
      isSpectrum={true}
      key="4"
      value={0}
    />
    <Block
      isSpectrum={true}
      key="5"
      value={0}
    />
    <Block
      isSpectrum={true}
      key="6"
      value={0}
    />
    <Block
      isSpectrum={true}
      key="7"
      value={0}
    />
    <Block
      isSpectrum={true}
      key="8"
      value={0}
    />
    <Block
      isSpectrum={true}
      key="9"
      value={0}
    />
  </View>
  <View
    key="14"
    style={
      Object {
        "flexDirection": "row",
        "justifyContent": "space-between",
        "width": 80,
      }
    }
  >
    <Block
      isSpectrum={true}
      key="0"
      value={0}
    />
    <Block
      isSpectrum={true}
      key="1"
      value={0}
    />
    <Block
      isSpectrum={true}
      key="2"
      value={0}
    />
    <Block
      isSpectrum={true}
      key="3"
      value={0}
    />
    <Block
      isSpectrum={true}
      key="4"
      value={0}
    />
    <Block
      isSpectrum={true}
      key="5"
      value={0}
    />
    <Block
      isSpectrum={true}
      key="6"
      value={0}
    />
    <Block
      isSpectrum={true}
      key="7"
      value={0}
    />
    <Block
      isSpectrum={true}
      key="8"
      value={0}
    />
    <Block
      isSpectrum={true}
      key="9"
      value={0}
    />
  </View>
  <View
    key="15"
    style={
      Object {
        "flexDirection": "row",
        "justifyContent": "space-between",
        "width": 80,
      }
    }
  >
    <Block
      isSpectrum={true}
      key="0"
      value={0}
    />
    <Block
      isSpectrum={true}
      key="1"
      value={0}
    />
    <Block
      isSpectrum={true}
      key="2"
      value={0}
    />
    <Block
      isSpectrum={true}
      key="3"
      value={0}
    />
    <Block
      isSpectrum={true}
      key="4"
      value={0}
    />
    <Block
      isSpectrum={true}
      key="5"
      value={0}
    />
    <Block
      isSpectrum={true}
      key="6"
      value={0}
    />
    <Block
      isSpectrum={true}
      key="7"
      value={0}
    />
    <Block
      isSpectrum={true}
      key="8"
      value={0}
    />
    <Block
      isSpectrum={true}
      key="9"
      value={0}
    />
  </View>
  <View
    key="16"
    style={
      Object {
        "flexDirection": "row",
        "justifyContent": "space-between",
        "width": 80,
      }
    }
  >
    <Block
      isSpectrum={true}
      key="0"
      value={0}
    />
    <Block
      isSpectrum={true}
      key="1"
      value={0}
    />
    <Block
      isSpectrum={true}
      key="2"
      value={0}
    />
    <Block
      isSpectrum={true}
      key="3"
      value={0}
    />
    <Block
      isSpectrum={true}
      key="4"
      value={0}
    />
    <Block
      isSpectrum={true}
      key="5"
      value={0}
    />
    <Block
      isSpectrum={true}
      key="6"
      value={0}
    />
    <Block
      isSpectrum={true}
      key="7"
      value={0}
    />
    <Block
      isSpectrum={true}
      key="8"
      value={0}
    />
    <Block
      isSpectrum={true}
      key="9"
      value={0}
    />
  </View>
  <View
    key="17"
    style={
      Object {
        "flexDirection": "row",
        "justifyContent": "space-between",
        "width": 80,
      }
    }
  >
    <Block
      isSpectrum={true}
      key="0"
      value={0}
    />
    <Block
      isSpectrum={true}
      key="1"
      value={0}
    />
    <Block
      isSpectrum={true}
      key="2"
      value={0}
    />
    <Block
      isSpectrum={true}
      key="3"
      value={0}
    />
    <Block
      isSpectrum={true}
      key="4"
      value={0}
    />
    <Block
      isSpectrum={true}
      key="5"
      value={0}
    />
    <Block
      isSpectrum={true}
      key="6"
      value={0}
    />
    <Block
      isSpectrum={true}
      key="7"
      value={0}
    />
    <Block
      isSpectrum={true}
      key="8"
      value={0}
    />
    <Block
      isSpectrum={true}
      key="9"
      value={0}
    />
  </View>
  <View
    key="18"
    style={
      Object {
        "flexDirection": "row",
        "justifyContent": "space-between",
        "width": 80,
      }
    }
  >
    <Block
      isSpectrum={true}
      key="0"
      value={0}
    />
    <Block
      isSpectrum={true}
      key="1"
      value={0}
    />
    <Block
      isSpectrum={true}
      key="2"
      value={0}
    />
    <Block
      isSpectrum={true}
      key="3"
      value={0}
    />
    <Block
      isSpectrum={true}
      key="4"
      value={0}
    />
    <Block
      isSpectrum={true}
      key="5"
      value={0}
    />
    <Block
      isSpectrum={true}
      key="6"
      value={0}
    />
    <Block
      isSpectrum={true}
      key="7"
      value={0}
    />
    <Block
      isSpectrum={true}
      key="8"
      value={0}
    />
    <Block
      isSpectrum={true}
      key="9"
      value={0}
    />
  </View>
  <View
    key="19"
    style={
      Object {
        "flexDirection": "row",
        "justifyContent": "space-between",
        "width": 80,
      }
    }
  >
    <Block
      isSpectrum={true}
      key="0"
      value={0}
    />
    <Block
      isSpectrum={true}
      key="1"
      value={0}
    />
    <Block
      isSpectrum={true}
      key="2"
      value={0}
    />
    <Block
      isSpectrum={true}
      key="3"
      value={0}
    />
    <Block
      isSpectrum={true}
      key="4"
      value={0}
    />
    <Block
      isSpectrum={true}
      key="5"
      value={0}
    />
    <Block
      isSpectrum={true}
      key="6"
      value={0}
    />
    <Block
      isSpectrum={true}
      key="7"
      value={0}
    />
    <Block
      isSpectrum={true}
      key="8"
      value={0}
    />
    <Block
      isSpectrum={true}
      key="9"
      value={0}
    />
  </View>
</View>
`;
